src/sidebar/components/HypothesisApp.tsx

```diff
@@ -155,12 +174,15 @@ function HypothesisApp({
           onSignUp={signUp}
           onLogout={logout}
           isSidebar={isSidebar}
+          showShareButton={shareDialogTabs.length > 0}
         />
       )}
       <div className="container">
         <ToastMessages />
         <HelpPanel />
-        <ShareDialog />
+        {isNonEmptyArray(shareDialogTabs) && (
+          <ShareDialog tabs={shareDialogTabs} />
```

Comment:
Since there are only three possible values for the tab, and we're not expecting to add new ones soon, you could just use three independent props to indicate which tabs to show. This would avoid the need for code to ensure the values are unique and to sort the tabs.

Reply:
I thought it would be weird that the component renders empty, hence enforcing at least one tab, and that's also why I didn't use individual props.

If we think this is not a problem, then individual props definitely simplify a couple of considerations. Will do the changes.